CMakeLists.txt

```diff
@@ -237,6 +245,9 @@ if(MSVC)
     endif()
 endif()
 
+configure_file(${Ginkgo_SOURCE_DIR}/include/ginkgo/batch_config.hpp.in
```

Comment:
Instead of all the extra code for a new config file, I would suggest you just query the value of `GINKGO_CUDA_BATCH_GMRES_USE_NO_SHMEM` in `cuda/CMakeLists.txt` and `add_definitions` to the `batch_gmres_kernels.cu` file.

Once we have two or three like this, we could either use a config file or add a cmake functions for the batched targets.